lib/raw_net_capture.rb

```diff
@@ -30,26 +44,27 @@ def initialize
   end
 
   def reset
-    @raw_received = StringIO.new
-    @raw_sent = StringIO.new
+    @transactions = [{
+      :received => StringIO.new,
+      :sent => StringIO.new
+    }]
   end
 
   def received(data)
-    @raw_received << data
+    @getting_response = true
+    transactions.last[:received] << data
   end
 
   def sent(data)
-    @raw_sent << data
-  end
-
-  def headers
-    separator = "\r\n\r\n"
-    raw_string = @raw_received.string
-    if headers_end_index = raw_string.index(separator)
-      raw_string[0...(headers_end_index + separator.length)]
-    else
-      raw_string
+    if @getting_response
+      @getting_response = false
+      @transactions << {
```

Comment:
Should be some way to DRY this up, same hash is created above. Maybe a class for `RawHTTPTransaction` or something?